vault/audit_broker.go

```diff
@@ -28,46 +29,61 @@ type AuditBroker struct {
 	sync.RWMutex
 	backends map[string]backendEntry
 	logger   log.Logger
+
+	broker *eventlogger.Broker
 }
 
 // NewAuditBroker creates a new audit broker
-func NewAuditBroker(log log.Logger) *AuditBroker {
+func NewAuditBroker(log log.Logger, useEventLogger bool) *AuditBroker {
+	var eventBroker *eventlogger.Broker
+
+	if useEventLogger {
+		// Ignoring the second error return value since an error will only occur
+		// if an unrecognized eventlogger.RegistrationPolicy is provided to an
+		// eventlogger.Option function.
+		eventBroker, _ = eventlogger.NewBroker(eventlogger.WithNodeRegistrationPolicy(eventlogger.DenyOverwrite), eventlogger.WithPipelineRegistrationPolicy(eventlogger.DenyOverwrite))
```

Comment:
This seems unwise and something that might come back to bite us later.  Why not just bite the bullet and add an error return val?

Reply:
In this case, I studied the code and the fact that it _can_ return an error is more or less due to the fact that the code is following a common pattern for option functions. However, this is code coming from another repository and therefore, we could get blindsided with a change in behaviour that would lead to errors being silently returned, I'll take you up on your suggestion.